pkg/discovery/helper.go

```diff
@@ -117,14 +119,21 @@ func (h *helper) Refresh() error {
 		return errors.WithStack(err)
 	}
 
+	preferredResources, err := refreshServerPreferredResources(h.discoveryClient, h.logger)
```

Comment:
I would only execute this if the feature flag is **not** set, since this call is only necessary in that case -- so you could put it in an `else` branch of the if statement on L129.

Reply:
Done!